### ISSUE TITLE
W1b: `infer_bbox_shape` / `bbox_to_mask` accept `(B, N, 4, 2)` and silently return wrong-shaped results when `N >= 3`

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4180]
## Summary

`infer_bbox_shape` (`kornia/geometry/bbox.py:112` at `e0969438`) and `bbox_to_mask` (`bbox.py:194`) index vertices at fixed positions on axis 1 (`boxes[:, 1, 0]`, `boxes[:, 2, 1]`, ...) and never check the rank. Both are documented for `(N, 4, 2)` input, but `Boxes.to_tensor('vertices_plus')` on a batched object returns `(B, N, 4, 2)`, and passing that straight through does one of two things depending on `N`:

- `N < 3`: `IndexError` (index 1 or 2 out of bounds on the box axis);
- `N >= 3`: **succeeds silently**, reading box 1 and box 2 instead of vertices 1 and 2 and returning values of the wrong shape.

`validate_bbox` in the same module flattens rank-4 input and accepts it, so a caller who validated first has no reason to expect the consumer to disagree.

## Reproduction

```python
import torch
from kornia.geometry.bbox import infer_bbox_shape, bbox_to_mask
from kornia.geometry.boxes import Boxes

for n in (1, 2, 3):
    x = torch.arange(n * 8, dtype=torch.float32).reshape(1, n, 4, 2)
    try:
        h, w = infer_bbox_shape(x)
        print(n, "infer_bbox_shape ->", tuple(h.shape), h.tolist(), w.tolist())
    except IndexError as e:
        print(n, "infer_bbox_shape ->", e)
    try:
        print(n, "bbox_to_mask ->", tuple(bbox_to_mask(torch.zeros(1, n, 4, 2), 5, 5).shape))
    except IndexError:
        print(n, "bbox_to_mask -> IndexError")

b = Boxes.from_tensor(torch.tensor([[[1., 2., 5., 4.], [0., 0., 2., 2.], [3., 3., 4., 4.]]]), "xyxy")   # (1, 3, 4)
print([t.tolist() for t in infer_bbox_shape(b.to_tensor("vertices_plus"))], "vs", [t.tolist() for t in b.get_boxes_shape()])
```

## Observed

```
1 infer_bbox_shape -> index 1 is out of bounds for dimension 1 with size 1
1 bbox_to_mask -> IndexError
2 infer_bbox_shape -> index 2 is out of bounds for dimension 1 with size 2
2 bbox_to_mask -> IndexError
3 infer_bbox_shape -> (1, 2) [[17.0, 17.0]] [[9.0, 9.0]]
3 bbox_to_mask -> (2, 5, 5)
[[[0.0, 2.0]], [[0.0, -1.0]]] vs [[[2.0, 2.0, 1.0]], [[4.0, 2.0, 1.0]]]
```

The last line is the realistic path: a batched `Boxes` with three boxes, exported and handed to `infer_bbox_shape`, yields a `(1, 2)` "heights" tensor of `[0, 2]` and "widths" of `[0, -1]` where `get_boxes_shape` says `(2, 2, 1)` and `(4, 2, 1)`. Nothing raises, and the shapes are plausible enough to flow into downstream code.

## Expected

Either accept `(B, N, 4, 2)` by flattening the leading axes the way `validate_bbox` does and returning `(B, N)` extents / `(B, N, H, W)` masks, or raise a `ShapeError`-style check on `boxes.ndim != 3` so the mistake is caught at the call site. Silent success with wrong shapes for `N >= 3` should not remain.

## Context

Found while reviewing #4060 (bbox/boxes convention blocks, batch 4a), which now states that rank-4 input is not supported by these two functions and tells `Boxes` users to index or flatten a batched export first. That PR is documentation only, so the shape check itself is out of scope there, and this issue is the anchor for that sentence. Whichever repair is chosen for #3934 (inclusive `+1`) touches the same two functions, so the check can ride along. Part of the #4020 audit.

## Environment

kornia 0.9.0rc1 @ `e0969438` (main), torch 2.9.1, python 3.11, cpu.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).

**Comment by hariomCodes** [comment]
I’d like to work on this. I plan to follow the currently documented rank-3 contract by adding an explicit shape check for rank-4 input in infer_bbox_shape and bbox_to_mask, with regression tests covering both the current IndexError cases and the silent wrong-shape case. I’ll keep the change focused and open a PR once the relevant tests pass.
